src/main/java/it/gov/pagopa/payhub/activities/service/treasury/TreasuryVersionBaseHandlerService.java

```diff
@@ -2,49 +2,103 @@
         this.validatorService = validatorService;
+        this.ingestionFlowFileArchiverService = ingestionFlowFileArchiverService;
     }
 
     abstract T unmarshall(File file);
 
     @Override
-    public Map<TreasuryOperationEnum, List<TreasuryDTO>> handle(File input, IngestionFlowFileDTO ingestionFlowFileDTO, int size) {
+    public Map<TreasuryOperationEnum, List<TreasuryDTO>> handle(File input, IngestionFlowFileDTO ingestionFlowFileDTO, int size, String errorDirectory) {
         try {
             T unmarshalled = unmarshall(input);
-            validate(ingestionFlowFileDTO, size, unmarshalled);
+            List<TreasuryErrorDTO> errorDTOList = validate(ingestionFlowFileDTO, size, unmarshalled);
             Map<TreasuryOperationEnum, List<TreasuryDTO>> result = mapperService.apply(unmarshalled, ingestionFlowFileDTO);
             log.debug("file flussoGiornaleDiCassa with name {} parsed successfully using mapper {} ", ingestionFlowFileDTO.getFileName(), getClass().getSimpleName());
+            writeErrors(errorDTOList, input.getName(), errorDirectory);
             return result;
         } catch (Exception e) {
             log.info("file flussoGiornaleDiCassa with name {} parsing error using mapper{} ", ingestionFlowFileDTO.getFileName(), getClass().getSimpleName());
             return Collections.emptyMap();
         }
     }
 
-    private void validate(IngestionFlowFileDTO ingestionFlowFileDTO, int size, T fGCUnmarshalled) {
+    List<TreasuryErrorDTO> validate(IngestionFlowFileDTO ingestionFlowFileDTO, int size, T fGCUnmarshalled) {
         if (validatorService.validatePageSize(fGCUnmarshalled, size)) {
             throw new TreasuryOpiInvalidFileException("invalid total page number for ingestionFlowFile with name " + ingestionFlowFileDTO.getFileName());
         }
-        validatorService.validateData(fGCUnmarshalled, ingestionFlowFileDTO.getFileName());
+        return validatorService.validateData(fGCUnmarshalled, ingestionFlowFileDTO.getFileName());
     }
+
+   void writeErrors(List<TreasuryErrorDTO> errorDTOList, String fileName, String errorDirectory) {
+
+       List<String[]> data = errorDTOList.stream()
+               .map(errorDTO -> new String[]{
+                       errorDTO.getNomeFile(),
+                       errorDTO.getDeAnnoBolletta(),
+                       errorDTO.getCodBolletta(),
+                       errorDTO.getErrorCode(),
+                       errorDTO.getErrorMessage()
+               })
+               .toList();
+
+
+       try {
+           String[] headerArray = new String[]{"FileName", "Anno Bolletta", "Codice Bolletta", "Error Code", "Error Message"};
+           List<String[]> header = new ArrayList<>(List.of());
+           header.add(headerArray);
+
+           String errorPathName = "ERROR-" + fileName;
+           CsvUtils.createCsv(errorPathName, header, data);
+
+           archiveErrorFile(new File(errorPathName), errorDirectory);
```

Comment:
errorDirectory would contains just the name of the directory, not the entire path, you should concatenate it to the ingestionFlowFile basePath (see how the original file is archived into its destination)